enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,609 @@
+    // that is defined in an existing CatalogSource resource in the cluster.
+    //
+    // This configured package will be managed with the cluster's lifecycle.
+    // +kubebuilder:validation:Required
+    PackageName string `json:"packageName"`
+}
+
+// PlatformOperatorStatus defines the observed state of PlatformOperator.
+type PlatformOperatorStatus struct {
+    // Conditions represent the latest available observations of a platform operator's current state.
+    // +optional
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+#### Upstream OLM 1.x API(s)
+
+> Note: The following API(s) will live in the openshift/operator-framework-rukpak repository.
+
+- [Introduces the upstream Bundle and BundleDeployment OLM 1.x v1alpha1 API(s)](https://github.com/operator-framework/rukpak/tree/main/api/v1alpha1).
+- [Introduces a validating webhook for the Bundle API](https://github.com/operator-framework/rukpak/blob/main/api/v1alpha1/bundle_webhook.go).
+- [Introduces a validation webhook for CRDs](https://github.com/operator-framework/rukpak/tree/main/cmd/crdvalidator).
+
+### Risks and Mitigations
+
+#### Risks
+
+- Early phases will introduce multiple v1alpha1 CRDs downstream.
+- Early phases will introduce OLM v1 components that are in alpha stages.
+- No concrete migration plan from legacy to OLM v1.
+- Coordination around external team dependencies.
+- The BundleDeployment API has a large net of permissions.
+- The upstream rukpak project has a direct dependency on cert-manager.
+
+#### Mitigations
+
+The v1alpha1 CRDs won't be deployed by default, and those APIs will be hidden behind the "TechPreviewNoUpgrade" feature set that admins need to explicitly enable before cluster creation. Any new components introduced during these early phases will also specify the `release.openshift.io/feature-gate` annotation in their CVO manifests, and set that key's value to "TechPreviewNoUpgrade".
+This annotation will ensure that the CVO won't deploy any tech preview components.
+
+Instructing admins to configure upfront RBAC to limit the BundleDeployment API to the minimum set of users may help mitigate the plain provisioner's wide net of permissions. The process for constructing this upfront RBAC may require explicit documentation.
+
+We can mitigate the upstream rukpak project's direct dependency on the cert-manager operator by removing that dependency in favor of the downstream service-ca-operator component.
+
+### Drawbacks
+
+- Implementing this PO mechanism using legacy OLM would provide more immediate business value. That implementation is outlined below in the alternatives section.
+- Early phases will be delivered through tech preview guidelines. This leads to little customer feedback, given it precludes production clusters.
+- Early phases may introduce a new OLM 1.x control plane that has little-to-no interaction with the existing, legacy OLM 0.x control plane.
```

Comment:
Is it worth expanding on this one?

One reason this is a drawback is because it can result in a sort of split-brain situation where incompatible operators end up being deployed since Rukpak and legacy OLM are not aware of what the other is doing.

Reply:
I think it's worth calling out that split brain behavior as a drawback, but there's still some ambiguity here on what that interaction looks like in practice given we haven't had any concrete examples of deploying packages using APIs from both control planes.